src/main/java/taxi/dao/impl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,195 @@
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultset = preparedStatement.executeQuery();
+            while (resultset.next()) {
+                cars.add(createCar(resultset));
+            }
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't getAll manufacturers ", ex);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET cars_model=?,manufacturers_id=? "
+                + " WHERE cars_id =? AND cars_deleted = 'false'";
+        String deleteQuery = "DELETE FROM car_drivers WHERE car_id = ?";
+        String newInsertionCar = "INSERT INTO car_drivers (driver_id, car_id) VALUES(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(updateQuery);
+                PreparedStatement deleteStatement = connection.prepareStatement(deleteQuery);
+                PreparedStatement newInsertionCarStatement = connection
+                        .prepareStatement(newInsertionCar)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+
+            deleteStatement.setLong(1, car.getId());
+            deleteStatement.executeUpdate();
+
+            newInsertionCarStatement.setLong(2, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                newInsertionCarStatement.setLong(1, driver.getId());
+                newInsertionCarStatement.executeUpdate();
+            }
+
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't update car " + car, ex);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String sqlQuery = "UPDATE cars"
+                + " SET cars_deleted = TRUE"
+                + " WHERE cars_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery)) {
+            preparedStatement.setLong(1, id);
+            int updatesMade = preparedStatement.executeUpdate();
+            return updatesMade > 0;
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't delete car by id " + id, ex);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM car_drivers "
+                + "JOIN drivers ON drivers_id=driver_id "
+                + "JOIN cars ON cars_id=car_id "
+                + "JOIN manufacturer ON manufacturer_id = manufacturers_id";
```

Comment:
let's check whether driver with this id is deleted and the same about cars

Reply:
done